extensions/windows-event-log/Bookmark.h

```diff
@@ -21,20 +22,26 @@ class Bookmark
 public:
   Bookmark(const std::wstring& channel, const std::wstring& query, const std::string& bookmarkRootDir, const std::string& uuid, bool processOldEvents, std::shared_ptr<core::CoreComponentStateManager> state_manager, std::shared_ptr<logging::Logger> logger);
   ~Bookmark();
-  operator bool() const;
-  EVT_HANDLE getBookmarkHandleFromXML();
+  explicit operator bool() const noexcept;
+
+  /* non-owning */ EVT_HANDLE getBookmarkHandleFromXML();
   bool getNewBookmarkXml(EVT_HANDLE hEvent, std::wstring& bookmarkXml);
   bool saveBookmarkXml(const std::wstring& bookmarkXml);
 private:
   bool saveBookmark(EVT_HANDLE hEvent);
   bool getBookmarkXmlFromFile(std::wstring& bookmarkXml);
 
+  struct evt_deleter {
+    void operator()(EVT_HANDLE) const noexcept;
+  };
+  using unique_evt_handle = std::unique_ptr<std::remove_pointer<EVT_HANDLE>::type, evt_deleter>;
```

Comment:
This `unique_evt_handle` could be useful in all Windows-related code; is there some "Windows utilities" header it could be moved to?

Reply:
I'm not aware of any. This extension is the only one that handles windows events, so I think the proper place for this utility is somewhere inside the extension subdirectory.

I think it should be extracted from the scope as soon as we need it elsewhere in the extension, but until then, I think it's better to keep it in a smaller scope.

Reply:
Agreed. Currently this is the only Wundows-specific codepart we have.

We might get back to create some Windows-untils in case further Win-specific features are needed.